### ISSUE TITLE
Broken with Guzzle 6

### ISSUE THREAD

**Issue by mhousser** [esbenp/lumen-api-oauth#2]
Guzzle 6 has a new validation step that results in the following error with this project:

```
InvalidArgumentException in Client.php line 380:

Passing in the "body" request option as an array to send a POST request has been deprecated. Please use the "form_params" request option to send a application/x-www-form-urlencoded request, or a the "multipart" request option to send a multipart/form-data request.
```

Appears to be triggered by the following in your Proxy.php:

```
$guzzleResponse = $client->post(sprintf('%s/oauth/access-token', $config->get('app.url')), [
    'body' => $data
]);
```

Full stack:

```
in Client.php line 380
at Client->invalidBody() in Client.php line 118
at Client->requestAsync('post', 'http://52.25.173.213/oauth/access-token', array('body' => array('client_id' => '1', 'client_secret' => 'gKYG75sw', 'grant_type' => 'password', 'username' => 'user@user.com', 'password' => '1234'), 'synchronous' => true)) in Client.php line 130
at Client->request('post', 'http://52.25.173.213/oauth/access-token', array('body' => array('client_id' => '1', 'client_secret' => 'gKYG75sw', 'grant_type' => 'password', 'username' => 'user@user.com', 'password' => '1234'))) in Client.php line 88
at Client->__call('post', array('http://52.25.173.213/oauth/access-token', array('body' => array('client_id' => '1', 'client_secret' => 'gKYG75sw', 'grant_type' => 'password', 'username' => 'user@user.com', 'password' => '1234')))) in Proxy.php line 39
at Client->post('http://52.25.173.213/oauth/access-token', array('body' => array('client_id' => '1', 'client_secret' => 'gKYG75sw', 'grant_type' => 'password', 'username' => 'user@user.com', 'password' => '1234'))) in Proxy.php line 39
at Proxy->proxy('password', array('username' => 'user@user.com', 'password' => '1234')) in Proxy.php line 12
at Proxy->attemptLogin(array('username' => 'user@user.com', 'password' => '1234')) in routes.php line 20
at Closure->{closure}()
at call_user_func_array(object(Closure), array()) in Container.php line 523
at Container->call(object(Closure), array()) in Application.php line 1197
at Application->callActionOnArrayBasedRoute(array(true, array(object(Closure)), array())) in Application.php line 1171
at Application->handleFoundRoute(array(true, array(object(Closure)), array())) in Application.php line 1093
at Application->Laravel\Lumen\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 141
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in AddQueuedCookiesToResponse.php line 36
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure)) in Pipeline.php line 125
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in OAuthExceptionHandlerMiddleware.php line 19
at OAuthExceptionHandlerMiddleware->handle(object(Request), object(Closure)) in Pipeline.php line 125
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 101
at Pipeline->then(object(Closure)) in Application.php line 1317
at Application->sendThroughPipeline(array('LucaDegasperi\OAuth2Server\Middleware\OAuthExceptionHandlerMiddleware', 'Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse'), object(Closure)) in Application.php line 1099
at Application->dispatch(null) in Application.php line 1037
at Application->run() in index.php line 28
```

Confirmed that downgrading to Guzzle 5.\* (via Composer dependency) fixes this problem.

**Comment by cesar-garay** [comment]
Hi, you can change body for form_params 'form_params' => $data
It worked for me.

**Comment by esbenp** [comment]
I will look into this ASAP. Thank you!

**Comment by esbenp** [comment]
Fixed in #5
